tritium-jmh/src/jmh/java/com/palantir/tritium/microbenchmarks/MetricsLifecycleBenchmark.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.microbenchmarks;
+
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.profile.GCProfiler;
+import org.openjdk.jmh.profile.JavaFlightRecorderProfiler;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@State(Scope.Benchmark)
+@SuppressWarnings({"designforextension", "NullAway", "PreferJavaTimeOverload"})
+public class MetricsLifecycleBenchmark {
+    private static final MetricName counterName =
+            MetricName.builder().safeName("counter").putSafeTags("tag", "value").build();
+    private static final MetricName histogramName = MetricName.builder()
+            .safeName("histogram")
+            .putSafeTags("tag", "value")
+            .build();
+    private static final MetricName meterName =
+            MetricName.builder().safeName("meter").putSafeTags("tag", "value").build();
+    private static final MetricName timerName =
+            MetricName.builder().safeName("timer").putSafeTags("tag", "value").build();
+    private TaggedMetricRegistry metrics;
+
+    @Setup
+    public void before() {
+        metrics = constructBaseRegistry();
+    }
+
+    @Benchmark
+    public void counter() {
+        metrics.counter(counterName).inc();
```

Comment:
I think we're better off avoiding interactions with the metric object since that's not directly related to lookup performance
```suggestion
    public Counter counter() {
        return metrics.counter(counterName);
```

Reply:
I was intending to have the benchmark capture more of the typical lookup & metric update so that we could compare optimizations from that perspective as opposed to just micro-optimizing the metric lookup/creation

Reply:
👍